### ISSUE TITLE
Release packaging failed against Elixir 1.4.2/Phoenix 13-rc.1 against Erlang 19.3

### ISSUE THREAD

**Issue by yeongsheng-tan** [bitwalker/distillery#224]
### Steps to reproduce

Phoenix 1.3-rc.1 umbrella app https://github.com/yeongsheng-tan/rna_umbrella/blob/master/README.md

Run `MIX_ENV=prod mix release --env=prod`

Was trying to thereafter use edib to build a docker container to run my phoenix umbrella app.
Ref guide from http://126kr.com/article/86nuobyth6t

### Verbose Logs

Paste the output of the release command you ran with the `--verbose` flag.

```
~/github/rna_umbrella master 6s
❯ MIX_ENV=prod mix release --env=prod --verbose
==> Loading configuration..
==> Assembling release..
==> Building release rna_umbrella:0.1.0 using environment prod
==> Discovered applications:
  snmp_mib_ex-0.0.3
    from: _build/prod/lib/snmp_mib_ex
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
    includes: none

  combine-0.9.6
    from: _build/prod/lib/combine
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  timex-3.1.13
    from: _build/prod/lib/timex
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :tzdata
      :gettext
      :combine
    includes: none

  phoenix_ecto-3.2.3
    from: _build/prod/lib/phoenix_ecto                                                                                                                                     [364/7374]
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :ecto
      :plug
    includes: none

  eex-1.4.2
    from: /usr/local/Cellar/elixir/1.4.2/bin/../lib/eex
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  phoenix-1.3.0-rc.1
    from: _build/prod/lib/phoenix
    applications:
      :kernel
      :stdlib
      :elixir
      :plug
      :poison
      :logger
      :eex
    includes: none

  mime-1.1.0
    from: _build/prod/lib/mime
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  plug-1.3.3
    from: _build/prod/lib/plug                                                                                                                                             [325/7374]
    applications:
      :kernel
      :stdlib
      :elixir
      :crypto
      :logger
      :mime
    includes: none

  phoenix_html-2.9.3
    from: _build/prod/lib/phoenix_html
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :plug
    includes: none

  cowlib-1.0.2
    from: _build/prod/lib/cowlib
    applications:
      :kernel
      :stdlib
      :crypto
    includes: none

  ranch-1.3.2
    from: _build/prod/lib/ranch
    applications:
      :kernel
      :stdlib
      :ssl
    includes: none

  cowboy-1.1.2
    from: _build/prod/lib/cowboy                                                                                                                                           [288/7374]
    applications:
      :kernel
      :stdlib
      :ranch
      :cowlib
      :crypto
    includes: none

  phoenix_pubsub-1.0.1
    from: _build/prod/lib/phoenix_pubsub
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :crypto
    includes: none

  poison-3.1.0
    from: _build/prod/lib/poison
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  gettext-0.13.1
    from: _build/prod/lib/gettext
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
    includes: none

  poolboy-1.5.1
    from: _build/prod/lib/poolboy
    applications:
      :kernel
      :stdlib
    includes: none

  ecto-2.1.4
    from: _build/prod/lib/ecto
    applications:
      :kernel                                                                                                                                                              [242/7374]
      :stdlib
      :elixir
      :logger
      :decimal
      :poolboy
    includes: none

  decimal-1.3.1
    from: _build/prod/lib/decimal
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  connection-1.0.4
    from: _build/prod/lib/connection
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  db_connection-1.1.2
    from: _build/prod/lib/db_connection
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :connection
    includes: none

  postgrex-0.13.2
    from: _build/prod/lib/postgrex
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :db_connection
      :decimal
      :crypto
    includes: none
                                                                                                                                                                           [197/7374]
  gpb-3.26.7
    from: _build/prod/lib/gpb
    applications:
      :kernel
      :stdlib
    includes: none

  exprotobuf-1.2.5
    from: _build/prod/lib/exprotobuf
    applications:
      :kernel
      :stdlib
      :elixir
      :gpb
    includes: none

  metrics-1.0.1
    from: _build/prod/lib/metrics
    applications:
      :kernel
      :stdlib
    includes: none

  ssl_verify_fun-1.1.1
    from: _build/prod/lib/ssl_verify_fun
    applications:
      :kernel
      :stdlib
      :ssl
    includes: none

  certifi-1.0.0
    from: _build/prod/lib/certifi
    applications:
      :kernel
      :stdlib
    includes: none

  mimerl-1.0.2
    from: _build/prod/lib/mimerl                                                                                                                                           [157/7374]
    applications:
      :kernel
      :stdlib
    includes: none

  idna-4.0.0
    from: _build/prod/lib/idna
    applications:
      :kernel
      :stdlib
    includes: none

  ssl-8.1.1
    from: /usr/local/Cellar/erlang/19.3/lib/erlang/lib/ssl-8.1.1
    applications:
      :crypto
      :public_key
      :kernel
      :stdlib
    includes: none

  public_key-1.4
    from: /usr/local/Cellar/erlang/19.3/lib/erlang/lib/public_key-1.4
    applications:
      :asn1
      :crypto
      :kernel
      :stdlib
    includes: none

  asn1-4.0.4
    from: /usr/local/Cellar/erlang/19.3/lib/erlang/lib/asn1-4.0.4
    applications:
      :kernel
      :stdlib
    includes: none

  crypto-3.7.3
    from: /usr/local/Cellar/erlang/19.3/lib/erlang/lib/crypto-3.7.3
    applications:
      :kernel
      :stdlib
    includes: none

  hackney-1.7.1
    from: _build/prod/lib/hackney
    applications:
      :kernel
      :stdlib
      :crypto
      :asn1
      :public_key
      :ssl
      :idna
      :mimerl
      :certifi
      :ssl_verify_fun
      :metrics
    includes: none

  tzdata-0.5.11
    from: _build/prod/lib/tzdata
    applications:
      :kernel
      :stdlib
      :elixir
      :hackney
      :logger
    includes: none

  calendar-0.17.2
    from: _build/prod/lib/calendar
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :tzdata
    includes: none

  quantum-1.9.0
    from: _build/prod/lib/quantum
    applications:
      :kernel
      :stdlib
      :elixir
      :calendar
    includes: none

  logger-1.4.2
    from: /usr/local/Cellar/elixir/1.4.2/bin/../lib/logger
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  compiler-7.0.4
    from: /usr/local/Cellar/erlang/19.3/lib/erlang/lib/compiler-7.0.4
    applications:
      :kernel
      :stdlib
    includes: none

  stdlib-3.3
    from: /usr/local/Cellar/erlang/19.3/lib/erlang/lib/stdlib-3.3
    applications:
      :kernel
    includes: none

  kernel-5.2
    from: /usr/local/Cellar/erlang/19.3/lib/erlang/lib/kernel-5.2
    applications: none
    includes: none

  elixir-1.4.2
    from: /usr/local/Cellar/elixir/1.4.2/bin/../lib/elixir
    applications:
      :kernel
      :stdlib
      :compiler
    includes: none

  iex-1.4.2
    from: /usr/local/Cellar/elixir/1.4.2/bin/../lib/iex
    applications:
      :kernel
      :stdlib
      :elixir
    includes: none

  sasl-3.0.3
    from: /usr/local/Cellar/erlang/19.3/lib/erlang/lib/sasl-3.0.3
    applications:
      :kernel
      :stdlib
    includes: none

  rna-0.0.1
    from: _build/prod/lib/rna
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :quantum
      :exprotobuf
      :postgrex
      :ecto
    includes: none

  rna_web-0.0.1
    from: _build/prod/lib/rna_web
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :gettext
      :poison
      :phoenix_pubsub
      :cowboy
      :phoenix_html
      :phoenix
      :rna
      :phoenix_ecto
    includes: none

  crontab-1.0.0
    from: _build/prod/lib/crontab
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :timex
    includes: none

  net_snmp_ex-0.0.32
    from: _build/prod/lib/net_snmp_ex
    applications:
      :kernel
      :stdlib
      :elixir
      :logger
      :snmp_mib_ex
    includes: none

==> Generated overlay vars:
    release_name=:rna_umbrella
    release_version="0.1.0"
    is_upgrade=false
    upgrade_from=:latest
    dev_mode=false
    include_erts=true
    include_src=false
    include_system_libs=true
    erl_opts=""
    erts_vsn="8.3"
    output_dir="_build/prod/rel/rna_umbrella"
==> Copying applications to _build/prod/rel/rna_umbrella
==> Generating nodetool
==> Generating start_erl.data
==> Generating vm.args
==> Generating sys.config from config/config.exs
==> Including ERTS 8.3 from /usr/local/Cellar/erlang/19.3/lib/erlang/erts-8.3
==> Generating boot script
==> Generating RELEASES
==> Generating start_clean.boot
==> Applying overlays
==> Applying mkdir overlay
    dst: releases/0.1.0/hooks
==> Applying mkdir overlay
    dst: releases/0.1.0/hooks/pre_configure.d
==> Applying mkdir overlay
    dst: releases/0.1.0/hooks/pre_start.d
==> Applying mkdir overlay
    dst: releases/0.1.0/hooks/post_start.d
==> Applying mkdir overlay
    dst: releases/0.1.0/hooks/pre_stop.d
==> Applying mkdir overlay
    dst: releases/0.1.0/hooks/post_stop.d
==> Applying mkdir overlay
    dst: releases/0.1.0/hooks/pre_upgrade.d
==> Applying mkdir overlay
    dst: releases/0.1.0/hooks/post_upgrade.d
==> Applying mkdir overlay
    dst: releases/0.1.0/commands
==> Packaging release..
==> Archiving rna_umbrella-0.1.0
==> Writing tarball to _build/prod/rel/rna_umbrella/releases/0.1.0/rna_umbrella.tar.gz
==> Release packaging failed due to errors:
    kernel: Cannot add file _build/prod/rel/rna_umbrella/lib/kernel-5.2/ebin/group_history.beam to tar file - _build/prod/rel/rna_umbrella/lib/kernel-5.2/ebin/group_history.beam: no such file or directory
```

### Description of issue

- What are the expected results?
```
==> Including ERTS 8.2.2 from /usr/local/Cellar/erlang/19.2.3/lib/erlang/erts-8.2.2
==> Packaging release..
==> Release successfully built!
    You can run it in one of the following ways:
      Interactive: _build/prod/rel/rna_umbrella/bin/rna_umbrella console
      Foreground: _build/prod/rel/rna_umbrella/bin/rna_umbrella foreground
      Daemon: _build/prod/rel/rna_umbrella/bin/rna_umbrella start
```
- What OS, Erlang/Elixir versions are you seeing this issue on?

```
macOS Sierra 10.12.3
MacBook Pro (Retina, 13-inch, Early 2015)
Processor 3.1 GHz Intel Core i7
Memory 16 GB 1867 MHz DDR3
Erlang 19.3
Elixir 1.4.2
Phoenix 1.3-rc.1
```

- If possible, also provide your `rel/config.exs`, as it is often
  my first troubleshooting question, and you'll save us both time :)

```
Path.join(["rel", "plugins", "*.exs"])
|> Path.wildcard()
|> Enum.map(&Code.eval_file(&1))

use Mix.Releases.Config,
    # This sets the default release built by `mix release`
    default_release: :default,
    # This sets the default environment used by `mix release`
    default_environment: Mix.env()

environment :dev do
  set dev_mode: true
  set include_erts: false
  set cookie: :"*mHU3UK9e$V5>!C^<_~q}(aTAK3]rK.nuuBbsH4f6SF)Py=*BKN3PjRMf5I;MtGH"
end

environment :prod do
  set include_erts: true
  set include_src: false
  set cookie: :"&7wB>_wmqrlDXnRru5a;qf{Kfk)@c2`6lT`&^t@L*^[gQ%O{?ccKOHZuSFQ2OT[n"
end

release :rna_umbrella do
  set version: "0.1.0"
  set applications: [
    rna: :permanent,
    rna_web: :permanent,
    crontab: :load,
    net_snmp_ex: :load,
    exprotobuf: :load
  ]
end
```

- Is there documentation that says one thing, but Distillery does
  another? If so, please link the doc here so it can be updated if
  it's a documentation issue, or so that the fix can be based around
  what's documented.

1. http://126kr.com/article/86nuobyth6t
2. https://github.com/bitwalker/distillery/blob/master/docs/Walkthrough.md
3. http://www.phoenixframework.org/docs/advanced-deployment

**Comment by bitwalker** [comment]
That's a really bizarre error - it's basically saying it can't tar up a file which it already copied to `_build`, which sounds like a potential permissions issue. Could you take a look at that path and verify that the file exists or not and what it's permissions are?

**Comment by yeongsheng-tan** [comment]
I did some more googling and trying to trace back what I last did before I hit the error.

I was finally able to nail down to erlang-history being the cause.

The instructions from [this link](https://gist.github.com/henrik/9e5250a83ccec6891f29e17f3f37277f) removed erlang-history for me, and allowed me to create the release build successfully.

Thank you for looking into this.

**Comment by yeongsheng-tan** [comment]
Lo and behold, erlang-history patched to support Erlang 19.3!!! I just did a git pull of git@github.com:ferd/erlang-history.git and saw the new patch commit:
```
~/github/erlang-history master
❯ gld
Commit: b7d436aac1ec37e88b7d455a44e15af36c36ef67 (HEAD -> master, origin/master, origin/HEAD)
Author: Fred Hebert <mononcqc@ferd.ca>
Date:   2017-03-16 08:43:58 -0400 (14 hours ago)

OTP-19.3 support
---
 Emakefile                   |   1 +
 README.markdown             |   2 +-
 ebin/5.2/.track-this-please |   0
 src/5.2/group.erl           | 876 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 src/5.2/group_history.erl   | 258 +++++++++++++++++++++++++++++++++++++++++++
 5 files changed, 1136 insertions(+), 1 deletion(-)

─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: Emakefile
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ Emakefile:1 @
{["src/5.2/*"], [debug_info, {outdir, "ebin/5.2"}]}.       % 19.3
{["src/5.1.1/*"], [debug_info, {outdir, "ebin/5.1.1"}]}.   % 19.2
{["src/5.1/*"], [debug_info, {outdir, "ebin/5.1"}]}.       % 19.1
{["src/5.0.1/*"], [debug_info, {outdir, "ebin/5.0.1"}]}.   % 19.0.3
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: README.markdown
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ README.markdown:92 @ DETS repairing of tables should be properly supported. In case of a database cor

### What Versions of Erlang/OTP does this work with? ###

I've tested it with all *quarterly* versions from R13B04 up to 19.2. It worked fine for them. I plan on supporting all newer *quarterly* versions. ALL OF THEM.
I've tested it with all *quarterly* versions from R13B04 up to 19.3. It worked fine for them. I plan on supporting all newer *quarterly* versions. ALL OF THEM.
```
